### PR TITLE
[PR] API Stage 2 - Adding tags

### PR DESCRIPTION
**Pull request by LuchoTurtle**
related to #256 

This PR adds tag support, alongside some refactoring related with [router helpers](https://elixirforum.com/t/router-helpers-with-same-controller-at-separate-scopes/44329).

### CODE REVIEW

**Comment by LuchoTurtle** [comment]
CRUD tags should now be properly working. 
The API tests have also been updated with this new introduction, as well as `seeds.exs`.

**Comment by LuchoTurtle** [comment]
Update on this.

I've been implementing the `tag` array to return proper errors. I've tried different approaches to this. I'm not sure whether I should *tell the user* that an item already exists or just ignore a tag that exists and add others that do not, and return a `200` successful request. 

I've implemented validators for both *checking if a tag already exists* and *if any tag is invalid*. However, the current implementation ignores tags that already exist. 

Additionally, I'm doing separate DB transactions for each `tag` instead of inserting all in the same so the [`insert_at` and `updated_at` fields are correctly setup](https://stackoverflow.com/questions/54197034/inserting-multiple-changesets-in-single-transaction). I'm aware I could pass it on manually but I think this makes the code more readable and simpler, as I don't have to manually add these fields this way.

**Comment by LuchoTurtle** [comment]
Went with telling the user a tag already exists and returning an error, instead of returning `200` and ignoring tags that already exist 👍

**Comment by LuchoTurtle** [comment]
Trying to implement the embedding of tags when retrieving items but I can't seem to idiomatically implement proper `json` encoding. 
I'm toggling `tag` loading depending on the `embed` query parameter passed on by the person using the API but I keep getting errors from `Jason` trying to encode it.

I've changed the `Jason.Encoder` annotation in the `item` schema to the following:
```elixir
  @derive {Jason.Encoder, except: [:__meta__, :__struct__, :timer, :inserted_at, :updated_at]}
```

But I keep getting this annoying error:
```sh
no function clause matching in Jason.Encoder.App.Item."-inlined-encode/2-"/2
```

*even* after deleting the `tag` key from the map.

I've tried the approaches of these three links but none helped :(

- https://elixirforum.com/t/better-jason-encoder-for-schemas-automatically/16245/12
- https://elixirforum.com/t/better-jason-encoder-for-schemas-automatically/16245/12
- https://elixirforum.com/t/runtimeerror-cannot-encode-association-address-from-myapp-model-property-to-json-because-the-association-was-not-loaded/39696/3

**Comment by LuchoTurtle** [comment]
I can't seem to get this to work, after countless attempts. I asked the question in Elixir Forums, hopefully I have an answer:

https://elixirforum.com/t/error-json-encoding-struct-no-function-clause-matching-in-jason-encoder-app-item-inlined-encode-2-2/53423

**Comment by nelsonic** [comment]
@LuchoTurtle thank you for giving a good account of your progress in this PR and opening the question on the Forum. 👍 
Let's discuss in-person tomorrow morning. 🙏

**Comment by LuchoTurtle** [comment]
Fetching items with tags is now possible. Tested and documented. Added new test suites to the CI, as well.

Going to check docs for any typos and then submit for review.

File: api.md [hunk 1]
```diff
@@ -28,19 +28,35 @@ can also be done through our `REST API`
...
+    - [7.3.1 Adding tests](#731-adding-tests)
+    - [7.3.2 Implementing `list_person_tags_text/0`](#732-implementing-list_person_tags_text0)
+    - [7.3.3 Updating `:create` in `lib/api/item.ex`](#733-updating-create-in-libapiitemex)
+      - [7.3.3.1 Creating `tag` validating functions](#7331-creating-tag-validating-functions)
+      - [7.3.3.2 Finishing up `lib/api/item.ex`'s `create` function](#7332-finishing-up-libapiitemexs-create-function)
+  - [7.3 Fetching items with embedded tags](#73-fetching-items-with-embedded-tags)
+    - [7.3.1 Adding tests](#731-adding-tests-1)
+    - [7.3.2 Returning `item` with `tags` on endpoint](#732-returning-item-with-tags-on-endpoint)
+    - [7.3.3 A note about `Jason` encoding](#733-a-note-about-jason-encoding)
+- [8. _Advanced/Automated_ `API` Testing Using `Hoppscotch`](#8-advancedautomated-api-testing-using-hoppscotch)
```
Comment (nelsonic) [inline api.md:54]: @LuchoTurtle please don't feel the need to keep pushing the `Hoppscotch` section down in this doc.
Renumbering is a tedious waste of time and does not actually reflect the order it was built in.
You can simply move this to it's own doc: `/lib/api/hoppscotch.md` and link to it from `API.md`.

File: lib/api/envs.json [hunk 2]
```diff
@@ -3,8 +3,8 @@
     "name": "Localhost",
     "variables": [
       {
-        "key": "host",
-        "value": "http://localhost:4000"
+        "value": "http://localhost:4000",
```
Comment (nelsonic) [inline lib/api/envs.json:6]: Is this file auto-generated? seems like this is change for the sake of change. 🤷‍♂️

**Review by nelsonic — APPROVED** [review]
Thanks @LuchoTurtle 👍

File: lib/api/envs.json (hunk 2 above)
Comment (LuchoTurtle) [reply lib/api/envs.json:6]: I'm exporting the environments everytime there's a change. So it's not "a change for the sake of change"

Comment (nelsonic) [reply lib/api/envs.json:6]: Sorry, not implying that you were the one making a change just for the sake of it. Just that the diff in this PR _appears_ that way. The variable key & value are the same.

**Pull request merged**